### PR TITLE
Enhance Device Control Response and Add AnyIoCtlCode

### PR DESCRIPTION
**Pull request by ibeckermayer**
- Introduced `AnyIoCtlCode` to handle cases with unknown or non-critical IoCtlCode.
- Updated `DeviceControlResponse` to accept an optional `output_buffer`, aligning with FreeRDP's implementation for empty buffer scenarios.
- Implemented decoding for `DeviceControlRequest<AnyIoCtlCode>` in `ServerDriveIoRequest`.

### CODE REVIEW

File: crates/ironrdp-rdpdr/src/pdu/efs.rs
Comment (CBenoit) [inline crates/ironrdp-rdpdr/src/pdu/efs.rs:2462]: Nitpick: we have `ensure_fixed_part_size!(in: src)`, specifically for this. I’ll not block the PR for that, feel free to send a follow up PR if you prefer.

**Review by CBenoit — COMMENTED** [review]
You need to rebase on top of master before we can merge (currently the PR includes commits from another branch already merged)

**Review by CBenoit — APPROVED** [review]
Approving for auto-merge once PR is rebased and conflicts are resolved.

File: crates/ironrdp-rdpdr/src/pdu/efs.rs
Comment (ibeckermayer) [reply crates/ironrdp-rdpdr/src/pdu/efs.rs:2462]: I just noticed that, and patched all those up in https://github.com/Devolutions/IronRDP/pull/298

**Comment by ibeckermayer** [comment]
> You need to rebase on top of master before we can merge (currently the PR includes commits from another branch already merged)

Oof, not quite sure what happened, I will make things make more sense.

**Pull request merged**
